zipkin/src/test/java/zipkin/storage/SpanStoreTest.java

```diff
@@ -828,6 +828,35 @@ public void clientTimestampAndDurationWinInSharedSpan() {
     }
   }
 
+  /**
+   * This test shows that if a parent ID is stored late (ex because it wasn't propagated), the span
+   * can resolve once it is.
+   */
+  @Test
+  public void lateParentIdInSharedSpan() {
```

Comment:
we don't merge client+server in java anymore. logic moved here https://github.com/openzipkin/zipkin/blob/master/zipkin-lens/src/zipkin/span-cleaner.test.js#L389